src/portfolios/portfolio/components/TaskOrder/TaskOrderDetails.vue

```diff
@@ -169,10 +395,211 @@ export default class TaskOrderDetails extends Vue {
+      const tableRowData: ClinTableRowData = {
+        isActive: isClinActive,
+        isExercised: clin.clin_status === Statuses.OptionExercised.value,
+        isPending: clin.clin_status === Statuses.OptionPending.value,
+        isExpired: clin.clin_status === Statuses.Expired.value,
+        CLINNumber: clin.clin_number,
+        CLINTitle: clin.idiq_clin_display?.display_value,
+        PoP: this.timeToExpiration(clin.pop_start_date,clin.pop_end_date),
+        popStartDate: clin.pop_start_date,
+        status: clin.clin_status,
+        statusLabel: getStatusLabelFromValue(clin.clin_status),
+        obligatedFunds: '$' + toCurrencyString(clin.funds_obligated),
+        totalCLINValue: '$' + toCurrencyString(clin.funds_total),
+        totalFundsSpent: '$' + toCurrencyString(clin.funds_spent_clin || 0),
+        isOverspent: clin.funds_spent_clin ? clin.funds_spent_clin > clin.funds_obligated : false,
+        fundsRemaining: this.fundsRemaining(
+          clin.funds_obligated,
+          String(clin.funds_spent_clin)
+        ),
+        startNewClinGroup: false
+      }
+     
+      //add row to appropriate temporary table
+      if (clin.clin_status === Statuses.Expired.value ){
+        this.expiredClins.push(tableRowData)
+      } else if (clin.clin_status === Statuses.OptionPending.value){
+        this.optionPendingClins.push(tableRowData)
+      } else {
+        this.tableData.push(tableRowData)
+      }
+
+      this.calculateTotalFundingObj(tableRowData);
+
+      this.totals = this.fundsRemaining(
+        this.totalFundingObj.totalObligatedFunds, this.totalFundingObj.totalFundsSpent
+      )
+    });
+    this.inactiveCount = this.expiredClins.length + this.optionPendingClins.length;
+
+  }
+
+  public async addSeparators() : Promise<void> {
+    this.optionPendingClins[0].startNewClinGroup = true;
+    this.expiredClins[0].startNewClinGroup = true;
+  }
+
+  public async sortRows() : Promise<void>{
+    this.tableData.sort((a, b) => (a.CLINNumber || "") > (b.CLINNumber || "") ? 1 : -1) 
+    this.sortArrayByDateThenNumber(this.optionPendingClins);
+    this.sortArrayByDateThenNumber(this.expiredClins);
+  }
+
+  public sortArrayByDateThenNumber(array: ClinTableRowData[]): void{
+    array.sort((a, b)=> {
+      if (a.popStartDate === b.popStartDate){
+        return (a.CLINNumber || "") > (b.CLINNumber || "") ? 1 : -1
+      } else {
+        return a.popStartDate > b.popStartDate ? -1 : 1
+      }
+    })
+  }
+
+  public async createTableData(): Promise<void>{
+    await this.collectTableData();
+    await this.sortRows();
+    await this.addSeparators();
+    /// create table data
+    this.tableData = [...this.tableData,...this.optionPendingClins, ...this.expiredClins]
+  }
+
+
+  public calculateTotalFundingObj(clin: ClinTableRowData): void{
+    if (clin.isActive || clin.isExercised) {
+      const totalCLINValue = currencyStringToNumber(clin.totalCLINValue || "0");
```

Comment:
A "TODO" to remove this line after feedback from the BA? Or a TODO to remove this function completely based on whether you need the below two lines?